oeps/oep-0026-arch-xapi-integration.rst

```diff
@@ -0,0 +1,483 @@
+
+As mentioned in `Using consistent user identifiers in Segment events`_, the other user
+identifiers in the Open edX platform aren't perfectly suited for inclusion in externally
+sent events:
+
+.. list-table::
+   :header-rows: 1
+
+   * - Identifier
+     - Limitations
+   * - username
+     - Often contains PII, such as user's name or year of birth.
+   * - LMS user_id
+     - Information leakage of auth_user table with easily guessable values; tied down
+       to implementation of the housing database.
+   * - email address
+     - Is PII and modifiable by the user.
+   * - anonymous user_id
+     - It is currently constructed by hashing the user's LMS user_id with the Django
+       server's *SECRET_KEY* value. This value will change when the *SECRET_KEY* is rotated.
+
+.. _Django auth_user: https://docs.djangoproject.com/en/2.0/topics/auth/default/#user-objects
+
+Example
+^^^^^^^
+
+Here is an example of an **Actor** JSON value that we would generate:
+
+::
+
+    "actor": {
+        “objectType”: “Agent”,
+        “openid”: “openedx.org/users/user-v1:<anonymized-user-id>”,
+        "name": “openedx.org/users/user-v1:<anonymized-user-id>”  # only include this field if necessary
+    }
+
+See `Deep Dive: Actor/Agent`_ for more information on xAPI Actors.
+
+.. _Friend of a Friend (FOAF): http://xmlns.com/foaf/spec/
+.. _email address: http://xmlns.com/foaf/spec/#term_mbox
+.. _hash of email address: http://xmlns.com/foaf/spec/#term_mbox_sha1sum
+.. _OpenID URI: http://xmlns.com/foaf/spec/#term_openid
+.. _account: http://xmlns.com/foaf/spec/#term_account
+.. _Personally Identifiable Information (PII): https://en.wikipedia.org/wiki/Personally_identifiable_information
+.. _Using consistent user identifiers in Segment events: https://openedx.atlassian.net/wiki/spaces/AN/pages/144441849/Using+consistent+user+identifiers+in+Segment+events
+.. _`Deep Dive: Actor/Agent`: https://xapi.com/deep-dive-actor-agent/
+
+xAPI Verb
+~~~~~~~~~
+
+The **Verb** in xAPI is a past tensed value, identified by a URI from the `xAPI registry`_
+and paired with a short display string. It denotes the action that was performed by the
+Actor on the Object in the statement. We will use only values found in the registry and
+document which values are used for which edX events in the `Open edX Events`_ section below.
+
+The registry is automatically created from multiple profiles. For now, we will limit
+ourselves to only URIs prefixed by the following domains, in the following priority
+order (in case of conflicting names):
+
+* http://adlnet.gov
+* http://w3id.org
+* http://id.tincanapi.com
+
+If, by any chance, a verb needed by Open edX does not exist in the registry, then we
+will create a pull request to recommend adding it to the `central github repository of
+xAPI Profiles`_. 
+
+Here is an example of a **Verb** JSON value that we would generate:
+
+::
+
+    "verb": {
+        "id": "http://adlnet.gov/expapi/verbs/answered",
+        "display": { 
```

Comment:
I am not sure that we need it - one of the problems with xAPI events is that each event is 2KB... if there are ways to minimize it and still be in the standard, thats great.

Reply:
Good point.  I update the spec to exclude the "display" field.